sdk/core/src/main/java/com/klaviyo/coresdk/model/UserInfo.kt

```diff
@@ -0,0 +1,103 @@
+package com.klaviyo.coresdk.model
+
+import com.klaviyo.coresdk.Registry
+import com.klaviyo.coresdk.model.ProfileKey.ANONYMOUS
+import com.klaviyo.coresdk.model.ProfileKey.EMAIL
+import com.klaviyo.coresdk.model.ProfileKey.EXTERNAL_ID
+import com.klaviyo.coresdk.model.ProfileKey.PHONE_NUMBER
+import java.util.UUID
+
+/**
+ * Stores information on the currently active user
+ */
+internal object UserInfo {
+
+    /**
+     * Save or clear an identifier in the persistent store and return it
+     *
+     * @param key
+     * @param value
+     * @return
+     */
+    private fun persist(key: ProfileKey, value: String): String = if (value.isEmpty()) {
+        Registry.dataStore.clear(key.name)
+    } else {
+        Registry.dataStore.store(key.name, value)
+    }.let { value }
+
+    /**
+     * Get value from persistent store or return a fallback if it isn't present
+     *
+     * @param key
+     * @param fallback
+     * @return
+     */
+    private fun fetch(key: ProfileKey, fallback: () -> String = { "" }): String =
+        Registry.dataStore.fetch(key.name).let { it.orEmpty().ifEmpty(fallback) }
+
+    var externalId: String = ""
+        set(value) { field = persist(EXTERNAL_ID, value) }
+        get() = field.ifEmpty { fetch(EXTERNAL_ID).also { field = it } }
+
+    var email: String = ""
+        set(value) { field = persist(EMAIL, value) }
+        get() = field.ifEmpty { fetch(EMAIL).also { field = it } }
+
+    var phoneNumber: String = ""
+        set(value) { field = persist(PHONE_NUMBER, value) }
+        get() = field.ifEmpty { fetch(PHONE_NUMBER).also { field = it } }
+
+    /**
+     * Anonymous ID is only used internally
+     * On first read, check for UUID in data store
+     * If not found, generate a fresh one and persist that
+     */
+    var anonymousId: String = ""
+        private set(value) { field = persist(ANONYMOUS, value) }
+        get() = field.ifEmpty { fetch(ANONYMOUS, generateUuid).also { anonymousId = it } }
+
+    private val generateUuid = { UUID.randomUUID().toString() }
+
+    /**
+     * Updates [UserInfo] identifiers in state from a given [Profile] object
+     *
+     * @param profile
+     */
+    fun updateFromProfile(profile: Profile) = apply {
+        externalId = profile.externalId ?: externalId
+        email = profile.email ?: email
+        phoneNumber = profile.phoneNumber ?: phoneNumber
+    }
+
+    /**
+     * Reset all user identifiers to defaults
+     * which will cause a new anonymous ID to be generated
+     */
+    fun reset() = apply {
+        externalId = ""
```

Comment:
This will call the setters above and do persistence stuff yah?

Reply:
Yup!